crates/interpreter/src/interpreter/shared_memory.rs

```diff
@@ -0,0 +1,297 @@
+use revm_primitives::U256;
+
+use crate::alloc::vec;
+use crate::alloc::vec::Vec;
+use core::{
+    cmp::min,
+    fmt,
+    ops::{BitAnd, Not},
+};
+
+/// A sequential memory shared between calls, which uses
+/// a `Vec` for internal representation.
+/// A [SharedMemory] instance should always be obtained using
+/// the `new` static method to ensure memory safety.
+pub struct SharedMemory {
+    /// Shared buffer
+    data: Vec<u8>,
+    /// Memory checkpoints for each depth
+    checkpoints: Vec<usize>,
+    /// Raw pointer used for the portion of memory used
+    /// by the current context
+    current_slice: *mut [u8],
+    /// How much memory has been used in the current context
+    current_len: usize,
```

Comment:
`*mut [u8]` (just like `&mut [u8]`) holds (ptr, len), so either use `*mut u8, usize` or just `*mut [u8]`

Reply:
Sorry that I misread/didn't understand it the first time, now it should be good!